test/string_view-test/string_view-test.cpp

```diff
@@ -0,0 +1,349 @@
+*	https://timsong-cpp.github.io/cppwp/n4659/string.view
+*/
+
+#include <stdexcept>
+
+#include "../verifiers.h"
+
+void string_view_test()
+{
+	using std::string_view;
+
+	//1. Construction and assignment
+	//2. Capacity
+
+	// Default constructor. Constructs an empty string_view. 
+	string_view empty_sv;
+	is_zero(empty_sv.size(), "empty_sv.size()");
+	is_true(empty_sv.length() == empty_sv.size(), "empty_sv.length() == empty_sv.size()");
+	is_true(empty_sv.empty(), "empty_sv.empty()");
+
+	//constructs a string_view with const char[]
+	const char c_hw[]{ "hello, world!" };
+	string_view sv(c_hw);
+	is_true(sv.size() == 13, "sv.size()");
+	is_true(sv.length() == sv.size(), "sv.length() == sv.size()");
+	is_false(sv.empty(), "!sv.empty()");
+
+	//Constructs a view of the first 5 characters of the character array
+	string_view f_sv(c_hw, 5);
+	is_true(f_sv.size() == 5, "f_sv.size()");
+	is_true(f_sv.length() == f_sv.size(), "f_sv.length() == f_sv.size()");
+	is_false(f_sv.empty(), "!f_sv.empty()");
+
+	//copy constructor
+	string_view c_sv(f_sv);
+	is_true(c_sv.size() == f_sv.size(), "c_sv.size() == f_sv.size()");
+	is_true(c_sv.length() == c_sv.size(), "c_sv.length() == c_sv.size()");
+	is_true(c_sv.empty() == f_sv.empty(), "c_sv.empty() == f_sv.empty()");
+
+	//assignment
+	string_view a_sv = c_sv;
+	is_true(a_sv.size() == c_sv.size(), "a_sv.size() == c_sv.size()");
+	is_true(a_sv.length() == a_sv.size(), "a_sv.length() == a_sv.size()");
+	is_true(a_sv.empty() == c_sv.empty(), "a_sv.empty() == c_sv.empty()");
+
+
+	//3. Non-member comparison functions
+	string_view ua("A");	//uppercase a
+	string_view la("a");	//lowercase a
+	string_view aua("A");	//another uppercase a
+	char csA[]{ "A" };		//C-string A
+	char csB[]{ "B" };		//C-string B
+
+	is_false(ua == la, "ua == la");
+	is_true(ua == aua, "ua == aua");
+	is_true(aua == ua, "aua == ua");
+	is_true(ua == csA, "ua == csA");
+	is_true(la != csB, "la != csB");
+	is_false(ua != aua, "ua != aua");
+	is_true(ua != la, "ua != la");
+	is_false(ua != csA, "!(ua != csA)");
+	is_true(ua != csB, "ua != csB");
+
+	is_false(ua > la, "ua > la");
+	is_true(ua < la, "ua > la");
+	is_true(la > ua, "la > ua");
+	is_false(la < ua, "!(la < ua)");
+	is_false(ua > aua, "!(ua > aua)");
+	is_false(ua < aua, "!(ua < aua)");
+	is_false(ua > csA, "!(ua > csA)");
+	is_false(ua < csA, "!(ua < csA)");
+	is_false(ua > csB, "!(ua > csB)");
+	is_true(ua < csB, "ua < csB");
+
+	is_false(ua >= la, "ua >= la");
+	is_true(ua <= la, "ua >= la");
+	is_true(la >= ua, "la >= ua");
+	is_false(la <= ua, "la >= ua");
+	is_true(ua >= aua, "ua >= la");
+	is_true(ua <= aua, "ua >= la");
+
+	is_true(ua >= csA, "ua >= la");
+	is_true(ua <= csA, "ua >= la");
+	is_false(ua >= csB, "ua >= la");
+	is_true(ua <= csB, "ua >= la");
+
+
+	//4. Element data and access
+	is_true(empty_sv.data() == nullptr, "empty_sv.data()");
+
+	//const char* str_1 = "hello, world!";
+	//default constructor and data test
+	string_view sv_1(c_hw);
```

Comment:
The comment says this tests the default constructor but I do not see the default constructor being tested.

Reply:
Thanks @Ndemco It's my mistake.